packages/teams-js/src/private/associatedApps.ts

```diff
@@ -0,0 +1,237 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ErrorCode, TabInstance } from '../public';
+import { runtime } from '../public/runtime';
+
+// Open questions
+// 1. I've re-used `TabInstance` from the public API, does that contain all of the information you and app developers might need?
+// 2. I didn't see any reason to add a `getTabs` function because `pages.tabs.getTabInstances`. Any reason that won't work for you?
+// 3. I've added an `AppTypes[]` param to `addAndConfigureApp` to allow for the host to show different app types to the user. Very open to changes.
+// 4. I've added empty, private `validate` functions for the threadId and TabInstance. Any validation that is possible will help prevent against
+//    bad data being sent to the host. If you have any validation that can be done, please add it there. If you *can* use restrictive types like UUID
+//    or something, that would be even better.
+// 5. I've made the namespace structure an empty `associatedApps` namespace that only contains the `tab` namespace. This was an attempt to leave room for
+//    expansion in the future for non-tab scenarios that will make it less likely that your callers will need to update their code. Open to opinions though.
+
+// TODO: Add unit tests
+// TODO: Add E2E tests
+
+/**
+ * @hidden
+ * @internal
+ * @beta
+ * Limited to Microsoft-internal use
+ *
+ * TODO: Brief description of what this capability does
+ */
+export namespace associatedApps {
+  export enum AppTypes {
+    meeting = 'meeting',
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * @beta
+   * Limited to Microsoft-internal use
+   *
+   * TODO: Brief description of what this capability does
+   */
+  export namespace tab {
+    interface ErrorResponse {
```

Comment:
Is there a reason you're not reusing SdkError?

Reply:
Nah, we can use that instead as long as we don't throw it directly, I can update that